hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestRPC.java

```diff
@@ -1107,6 +1108,45 @@ public TestRpcService run() {
     }
   }
 
+  @Test
+  public void testNumInProcessHandlerMetrics() throws Exception {
+    UserGroupInformation ugi = UserGroupInformation.
+      createUserForTesting("user123", new String[0]);
+    // use 1 handler so the callq can be plugged
+    final Server server = setupTestServer(conf, 1);
+    try {
+      RpcMetrics rpcMetrics = server.getRpcMetrics();
+      assertEquals(rpcMetrics.getNumInProcessHandler(), 0);
+
+      ExternalCall<String> call1 = newExtCall(ugi,
+          new PrivilegedExceptionAction<String>() {
+        @Override
+        public String run() throws Exception {
+          assertTrue(rpcMetrics.getNumInProcessHandler() > 0);
```

Comment:
How about exact comparison? We can use `assertEquals(1, rpcMetrics.getNumInProcessHandler())`

Reply:
Thank you @virajjasani for your comments and reviews.
I will update it later.